cartoframes/data/observatory/catalog/entity.py

```diff
@@ -123,21 +117,15 @@ def _download(self, file_path, credentials):
 
         project, dataset, table = full_remote_table_name.split('.')
 
-        try:
-            column_names = bq_client.get_table_column_names(project, dataset, table)
-            query = 'SELECT * FROM `{}`'.format(full_remote_table_name)
-            job = bq_client.query(query)
+        column_names = bq_client.get_table_column_names(project, dataset, table)
+        query = 'SELECT * FROM `{}`'.format(full_remote_table_name)
+        job = bq_client.query(query)
 
-            bq_client.download_to_file(job, file_path, column_names=column_names)
-        except NotFound:
-            raise CartoException('You have not purchased the dataset `{}` yet'.format(self.id))
+        bq_client.download_to_file(job, file_path, column_names=column_names)
```

Comment:
Same here, 

You can have access to DO, but not to a Dataset or Geography. How are we checking that?

Reply:
This is checked also in the `do/subscriptions` call, after the download.

Reply:
Yes, you are right. Since we are checking if the user is subscribed, ideally we can avoid the try-except. In any case, I will keep it as I tell you offline (but this is opinionable)